shallows/library/known.go

```diff
@@ -29,6 +29,12 @@ func KnownImportedUintID(prefix string, id uint64) string {
 	return fmt.Sprintf("%x-0000-0000-%04x-%012x", fnv.New32().Sum([]byte(prefix))[:4], h, l)
 }
 
+// create a unique import id from a uuid by mutating its first 4 bytes with the prefix checksum.
+func KnownImportedUUID(prefix string, id uuid.UUID) uuid.UUID {
```

Comment:
🤔  not necessarily a review comment.
```suggestion
func KnownImportedUUID[T ~string|~[]byte](source T, id uuid.UUID) uuid.UUID {
```